client/src/screens/PoiDetails/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect} from 'react';
+import {NavigationScreenComponent} from 'react-navigation';
+import {useQuery} from 'react-apollo-hooks';
+import {gql} from 'apollo-boost';
+import _ from 'lodash';
+import * as Types from '../../types/graphql';
+import {FooterButton, LoadingScreen, WindowHeader} from '../../components';
+import {Routes, env} from '../../consts';
+
+import {
+    Container,
+    Scroll,
+    Picture,
+    Content,
+    Title,
+    Subtitle,
+    Description,
+} from './atoms';
+
+const GET_POI = gql`
+    query Poi($id: ID!) {
+        poi(id: $id) {
+            name
+            description
+            photos {
+                content {
+                    url
+                }
+            }
+        }
+    }
+`;
+
+export type PoiDetailsScreenParams = {
+    poiId: Types.Poi['id'],
+};
+
+export const PoiDetailsScreen: NavigationScreenComponent<PoiDetailsScreenParams> = ({navigation}) => {
+    const id = navigation.getParam('poiId');
+
+    const {loading, data, error} = useQuery<Types.Query>(GET_POI, {
+        variables: {id},
+    });
+
+    useEffect(() => {
```

Comment:
Are you sure it suits here? You'll probably do one more render with an error

Reply:
Calling outside function is a side effect so it's in `useEffect`. Once `error` is changed, the callback is called without rerender. At least it is how I think it works :D

Reply:
it would work as cDU in the current case https://reactjs.org/docs/hooks-effect.html#explanation-why-effects-run-on-each-update